src/stable.js

```diff
@@ -14,9 +16,20 @@ function thunk(fn) {
     if (evaluated) {
       return result;
     } else {
-      result = fn();
+      result = fn.call(this);
       evaluated = true;
       return result;
     }
   };
 }
+
+export function cacheGetters(descriptors) {
+  return keys(descriptors).reduce(function(memo, key) {
```

Comment:
Let's just use `map` here, the structure of the output descriptors should be the same as the input descriptors, so a fold is overkill. Also, I'd say just put this right in `semigroup/object` until we need it later.

Reply:
@cowboyd I can't use map here because map uses append. It's causes an infinite loop.